docs/proposals/secret-annotation-new-app.md

```diff
@@ -0,0 +1,60 @@
+Design proposal to implement [Use build source secret based on annotation on
+Secret](https://trello.com/c/NoVpS2OS/1059-5-use-build-source-secret-based-on-annotation-on-secret-builds)
+
+(The card goal is for oc new-app/new-build to discover a relevant pre-existing
+Secret and attempt to use it when a source repo requires authentication.  The
+intention is to do this without adding yet more CLI arguments to
+new-app/new-build.)
+
+
+Design proposal:
+----------------
+
+- A Secret may include an annotation with a well-known name.  The name
+  *io.openshift.match-urls* is proposed.
+
+- A valid annotation value consists of zero or more comma-separated
```

Comment:
> Several examples of the comma-separated approach already exist elsewhere in our API as well. FWIW I personally think the end result of multiple inserts and deletes is likely to be much easier to comprehend if it's a single comma-separated field.

Can you express the `oc` command required to add an *additional* value the annotation for the command delimited case.  The prefix case is `oc annotate secret/foo build.openshift.io/clone-url.localgit https://foo`.